internal/services/oidc.go

```diff
@@ -106,33 +121,67 @@ func (osvc *OIDCService) IsFlowSecure(state string, token models.Tokens) (bool,
 	return true, nil
 }
 
-// CreateJWT creates a new token and the claims you would like it to contain.
-func (osvc *OIDCService) CreateJWT(cc *models.CustomClaims) (models.CustomClaims, error) {
-	// For HMAC signing method, the key can be any []byte
-	hmacRandSecret, err := randomByte(1990)
-	if err != nil {
-		osvc.logger.Errorf("An error occurred while generating HMAC. Error: %v", err)
-		return models.CustomClaims{}, err
-	}
+// Create a new JWT token based on Custom Claims models.
+func (osvc *OIDCService) Create(cc *models.CustomClaims) (models.CustomClaims, error) {
+	var err error
 
 	customClaims := models.CustomClaims{
 		Name:    cc.Name,
 		Email:   cc.Email,
 		Profile: cc.Profile,
 		RegisteredClaims: jwt.RegisteredClaims{
-			ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Duration(osvc.client.Configuration.JWTExpires))),
+			ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Duration(osvc.configuration.UserJWTExpires))),
 			IssuedAt:  jwt.NewNumericDate(time.Now()),
 		},
 	}
 
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, customClaims)
 
 	// Sign and get the complete encoded token as a string using the secret.
-	customClaims.AccessToken, err = token.SignedString(hmacRandSecret)
+	customClaims.AccessToken, err = token.SignedString(osvc.configuration.HMACSecret)
 
 	return customClaims, err
 }
 
+// Decode validates the token and returns the claims.
+func (osvc *OIDCService) Decode(tokenString string) (models.TokenInfo, error) {
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		// Don't forget to validate the alg is what you expect:
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+
+		return osvc.configuration.HMACSecret, nil
+	})
+
+	if token == nil {
+		return models.TokenInfo{}, errors.New("an unexpected error occurred while validating the JWT token")
+	}
+
+	if token.Valid {
+		claims := token.Claims.(jwt.MapClaims)
+		return models.TokenInfo{
+			UID:  fmt.Sprintf("%s%s", osvc.configuration.UserIDPrefix, claims[osvc.configuration.UserIDClaim].(string)),
```

Comment:
Do you know why this prefix is needed?

Reply:
yeap, I believe we'll use email.
but in some cases, you (or who will use this project) can configure the UID to be anything else and have something like a domain, e.g: `domain/username`... stuff like that.

Reply:
But I'm asking about the prefix usage, not the UID. Why do we need that inside the token, the userID isn't enough?

Reply:
this was used before we migrate to dex, when we used google iap straight, without something like dex.

i understand why did you put this, although i don't think we will use it anymore.

i believe we can remove it